scopes/ui-foundation/sidebar/ui/side-bar/side-bar.tsx

```diff
@@ -10,15 +10,18 @@ export type SideBarProps = {
    * slot of registered drawers.
    */
   drawerSlot: DrawerSlot;
-
+  /**
+   * slot of registered items to the main section at the top.
+   */
   itemSlot?: SidebarItemSlot;
 } & React.HTMLAttributes<HTMLDivElement>;
 
 /**
  * side bar component.
  */
 export function SideBar({ drawerSlot, itemSlot, ...rest }: SideBarProps) {
-  const [openDrawerList, onToggleDrawer] = useState([drawerSlot.toArray()[0][0]]);
+  const drawers = flatten(drawerSlot.values());
+  const [openDrawerList, onToggleDrawer] = useState([drawers[0].id]);
```

Comment:
make sure to handle undefined, ie `drawers[0]?.id`. Make sure the type is possibly undefined